File: examples/python/01_product.py
"""Fetch one Amazon product through Glade API."""

from glade_client import get


payload = get(
    "/api/amazon/product",
    {"asin": "B0D1XD1ZV3", "domain": "US"},
)
product = payload["data"]["amazonProduct"]
price = product.get("price") or {}

print(f"{product['title']} ({product['asin']})")
print(f"Brand: {product.get('brand') or 'n/a'}")
print(f"Price: {price.get('display') or 'n/a'}")
print(f"Rating: {product.get('rating') or 'n/a'}")
print(f"Prime: {product.get('isPrime')}")

File: examples/node/01-product.mjs
import { get } from "./glade-client.mjs";

const payload = await get("/api/amazon/product", {
  asin: "B0D1XD1ZV3",
  domain: "US",
});
const product = payload.data.amazonProduct;

console.log(`${product.title} (${product.asin})`);
console.log(`Brand: ${product.brand ?? "n/a"}`);
console.log(`Price: ${product.price?.display ?? "n/a"}`);
console.log(`Rating: ${product.rating ?? "n/a"}`);
console.log(`Prime: ${product.isPrime ?? "n/a"}`);
